climada/engine/impact.py

```diff
@@ -1389,6 +1389,114 @@ def _selected_exposures_idx(self, coord_exp):
             LOGGER.warning("No exposure coordinates match the selection.")
         return sel_exp
 
+    @classmethod
+    def concat(cls, imp_list, reset_event_ids = False):
+        """
+        Concatenate impact objects with the same exposure.
+        This function is useful if, e.g. different impact functions
+        have to be applied for different seasons (e.g. for agricultural impacts).
+
+        It concatenates the following attributes:
+            event_id, event_name, date, frequency, imp_mat, at_event, eai_exp, aai_exp
+        and takes the following attributes from the first impact object in the passed imp_list
+            coord_exp, crs, unit, tot_value, tag
+
+        NOTES
+        -----
+        - Concatenation of impacts with different exposure (e.g. different countries)
+                could also be implemented here in the future.
+
+        Parameters
+        ----------
+        imp_list : list of climada.engine.Impact objects
+            list of Impact objects to concatenate
+        reset_event_ids: boolean, optional
+            reset event ids of the concatenated impact object
+
+        Returns
+        --------
+        impact: climada.engine.Impact
+            New impact object which is a concatenation of all impacts
+        """
+        def check_exposure(imp_list):
+            """ checks if impacts are based on the same exposure """
+
+            #check crs
+            crs = {imp.crs for imp in imp_list}
+            if len(set(crs)) > 1:
+                raise ValueError("The impacts have different exposure crs.")
+
+            #check total value
+            tot_vals = np.unique([imp.tot_value for imp in imp_list])
+            if len(tot_vals) > 1:
+                raise ValueError("The impacts have different total exposure values.")
+
+            # compare exposure coordinates
+            for imp2 in imp_list[1::]:
+                if not np.array_equal(imp_list[0].coord_exp, imp2.coord_exp):
+                    raise ValueError("The impacts have different exposure coordinates.")
+
+            return True
+
+        #check if exposure is consistent
+        check_exposure(imp_list)
+
+        # concatenate event IDs
+        event_ids = [event_id for imp in imp_list for event_id in imp.event_id]
+        if reset_event_ids:
+            event_id = np.array(list(range(len(event_ids))))
+        else:
+            if len(event_ids) != len(set(event_ids)):
+                raise ValueError("Duplicate event IDs found.\
+                Consider to set reset_event_id = True.")
+            event_id = np.array(event_ids)
+
+        # concatenate impact matrices
+        imp_mats =[imp.imp_mat for imp in imp_list]
+        if None in imp_mats:
+            raise ValueError("Passed an impact object with an undefined impact matrix.")
```

Comment:
I just checked this: The [Impact constructor](https://github.com/CLIMADA-project/climada_python/blob/develop/climada/engine/impact.py#L190) sets the `imp_mat = sparse.csr_matrix(np.empty((0, 0)))` if no impact matrix is passed, so this `Impact.imp_mat` attribute should never be `None`!
```suggestion
```